examples/Example11/example11.cu

```diff
@@ -320,8 +321,20 @@ void example11(const vecgeom::cxx::VPlacedVolume *world, int numParticles, doubl
               << " number of hits: " << std::setw(4) << stats->scoring.hits;
     std::cout << std::endl;
 
+    // Check if only charged particles are left that are looping.
+    numElectrons = stats->inFlight[ParticleType::Electron];
+    numPositrons = stats->inFlight[ParticleType::Positron];
+    numGammas = stats->inFlight[ParticleType::Gamma];
+    if (numElectrons == previousElectrons && numPositrons == previousPositrons && numGammas == 0) {
+      loopingNo++;
+    } else {
+      previousElectrons = numElectrons;
+      previousPositrons = numPositrons;
+      loopingNo = 0;
+    }
+
     iterNo++;
-  } while (inFlight > 0 && iterNo < 1000);
+  } while (inFlight > 0 && loopingNo < 20);
```

Comment:
This is mostly fine because there are no secondaries produced, but we should deposit the energy of the loopers. A way is to setup a kill signal visible in the kernels, then doe the killing there.

Reply:
And we maybe we should account for the "killed" energy to tune the 20 value

Reply:
TestEm3 has this killing, but I plan to remove it because Geant4 doesn't deposit energy of killed particles either.

> And we maybe we should account for the "killed" energy to tune the 20 value

First, getting the sum of "killed" energy requires a reduction, which is expensive on GPUs. But what should we do with the value? Is 1 MeV ok or not? And what if the particles continue to loop while they are above a threshold?

Reply:
Well, it is all about making sure the errors we do by this procedure are acceptable and what category of loopers we kill. I mean, killing slow curly electrons may be OK, but killing particles having geometry errors may require a different action. Just saying it's good to understand what we kill, depending on that depositing the energy in care there is a sensitive volume may or may not be needed. Did you have a look on the kinematic properties for the loopers?

Reply:
I have looked at the deposited energy, and that remains exactly the same as before. Plus Geant4 doesn't deposit energy of looping particles either.

Reply:
Then we are fine for now, but there may be cases in future where this simple approach can introduce non-negligible bias, and we need to keep in mind than the number 20 used now may not be ideal in other cases.